packages/commonwealth/client/scripts/views/components/react_quill_editor/react_quill_editor.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import ReactQuill from 'react-quill';
+import type { DeltaStatic } from 'quill';
+import type { QuillMode } from '../quill/types';
+
+type ReactQuillEditorProps = {
+  className?: string
+  placeholder?: string;
+  tabIndex?: number;
+  theme?: string;
+  mode?: QuillMode; // Use in order to limit editor to only MD or RT support
+  contentDelta: DeltaStatic
+  setContentDelta: (d: DeltaStatic) => void
+  onChange?: (v: string) => void
+}
+
+// ReactQuillEditor is a custom wrapper for the react-quill component
+export function ReactQuillEditor(props: ReactQuillEditorProps) {
+
+  const {
+    className = '',
+    placeholder,
+    tabIndex,
+    theme,
+    contentDelta,
+    setContentDelta,
+    onChange
+  } = props
+
+  return (
+    <ReactQuill
+      className={`QuillEditor ${className}`}
+      placeholder={placeholder}
+      tabIndex={tabIndex}
+      theme={theme}
+      value={contentDelta}
+      onChange={(value, delta, source, editor) => {
+        console.log('value:', value)
+        setContentDelta(editor.getContents())
+        onChange?.(value)
+      }}
+    />
+  )
+
+}
```

Comment:
We are gonna write some react styleguide but generally let's go this way:
1. const instead of function
2. destructuring props right away
3. creating handlers above the return to not pollute view area with the business logic

```suggestion
const ReactQuillEditor = ({
    className = '',
    placeholder,
    tabIndex,
    theme,
    contentDelta,
    setContentDelta,
    onChange
  } : ReactQuillEditorProps) => {
  
  const handleChange = (value, delta, source, editor) => {
        console.log('value:', value)
        setContentDelta(editor.getContents())
        onChange?.(value)
      }
  
  return (
    <ReactQuill
      className={`QuillEditor ${className}`}
      placeholder={placeholder}
      tabIndex={tabIndex}
      theme={theme}
      value={contentDelta}
      onChange={handleChange}
    />
  )

}

export default ReactQuillEditor
```